### PR TITLE
docs: improve documentation related to File templates feature

### LINKED ISSUE #1199: [📑 Docs]: Improve documentation related to `File templates` feature
### What Dev Docs changes are you proposing?

- we need to make it clear that https://github.com/asyncapi/generator/blob/master/apps/generator/docs/file-templates.md is a feature for Nunjucks render engine only
- we need a new document, or refactor the one about `File templates` and explain how to do the same in react

we have at least comparizon from the perspective of "template for templates". So we see file templates in use in https://github.com/asyncapi/template-for-generator-templates/tree/nunjucks/template/schemas in times of nunjucks and then after conversion to react https://github.com/asyncapi/template-for-generator-templates/blob/master/template/schemas/schema.js

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Florence-Njeri**
**Description**
Improve documentation related to File templates feature:
- Separate nunjucks and react render engine file template docs
- Link a real life example of react render engine file template in use

**Related issue(s)**
Resolves #1199

### CODE REVIEW

**Comment by aeworxet** [comment]
@asyncapi/bounty_team

File: apps/generator/docs/file-templates.md
Comment (lmgyuan) [inline apps/generator/docs/file-templates.md:1]: Is it a typo to have 2 "refer to the"?

Comment (Florence-Njeri) [reply apps/generator/docs/file-templates.md:1]: Hey @lmgyuan, Thank you for pointing it out,  I have fixed it

**Comment by Gmin2** [comment]
LGTM

**Review by Gmin2 — COMMENTED** [review]
LGTM

just one point you can add more examples

**Comment by derberg** [comment]
@Gmin2 please suggest one

**Comment by derberg** [comment]
@Gmin2 kind reminder

**Review by Gmin2 — COMMENTED** [review]
There could be an example of generating with varied content based on different components

```jsx
import { File } from '@asyncapi/generator-react-sdk';

export default function({ asyncapi }) {
  const files = [];

  // Generate files for channels
  asyncapi.channels().forEach((channel, channelName) => {
    files.push(
      <File name={`channels/${channelName}.md`}>
        # Channel: {channelName}

        Description: {channel.description() || 'No description provided'}

        ## Operations
        {channel.publish() && <div>Publish: {channel.publish().summary()}</div>}
        {channel.subscribe() && <div>Subscribe: {channel.subscribe().summary()}</div>}
      </File>
    );
  });
  return files
  }
```

**Comment by Florence-Njeri** [comment]
Hey @Gmin2 I have updated the PR

**Review by Gmin2 — APPROVED** [review]
LGTM

**Comment by derberg** [comment]
folks we need an example that is valid. Provided example is not valid and looks like generated with AI, as it is inventing not existing features and using old API.

it is critical to always run examples manually before they are added to docs. Since we have https://github.com/asyncapi/generator/tree/master/apps/generator/test/test-templates/react-template/template in the generator now, it should be as simple as locally adding new file with an example, then you run the generation with CLI and see the result

here is a simplified version of the example:

```
import { File, Text } from "@asyncapi/generator-react-sdk";

export default function ({ asyncapi }) {
  const files = [];

  // Generate files for channels
  asyncapi.channels().forEach((channel) => {
    const channelName = channel.id();

    files.push(
      <File name={`${channelName}.md`}>
        <Text newLines={2}># Channel: {channelName}</Text>
        <Text>
          {channel.hasDescription() && `${channel.description()}`}
        </Text>
      </File>
    );
  });
  return files;
}
```

it shows:
- how multiline files generation must be handled with `Text` component, otherwise all gets generated as one line
- it shows how to work with new API of the parser - more API details in https://github.com/asyncapi/parser-api/blob/master/docs/api.md
- `<File name={`channels/${channelName}.md`}>` will not work, you will get an error. If you want to generate channel MD files in `channels` directory, you need to put template file in `channels` directory

@Gmin2 your example is good from idea point of view, so now just please take my sample end extend with part where you render operations information (forget about publish and subscribe, it is not there in new AsyncAPI anymore)

**Comment by Florence-Njeri** [comment]
Hey @derberg @Gmin2 I have updated the PR with the up-to-date code snippet. Please help me re-review. Thank you!

File: apps/generator/docs/file-templates.md [hunk 1]
```diff
@@ -64,4 +74,57 @@ export default function({ asyncapi }) {
...
+In practice, to render the multiple files, that are generated from the data defined in your AsyncAPI, you'll iterate over the array of schemas and generate a file for each schema as shown in the example below:
+
+```js
+/*
+ * To render multiple files, it is enough to return an array of `File` components in the rendering component, like in following example.
+ */
+export default function({ asyncapi }) {
+  const schemas = asyncapi.allSchemas();
+  // schemas is an instance of the Map
+  return Array.from(schemas).map(([schemaName, schema]) => {
```
Comment (derberg) [inline apps/generator/docs/file-templates.md:90]: schemas is not an array but a map, so like I suggest in my previous comment to the PR, better use forEach

and yeah, you need to manually test if file is working, as this way you would know something is wrong

during generation I get `Generator Error: object is not iterable (cannot read property Symbol(Symbol.iterator))`

btw, this example also needs `import { File, Text } from "@asyncapi/generator-react-sdk";`

**Comment by Florence-Njeri** [comment]
Hey @derberg I fixed the code

**Review by derberg — APPROVED** [review]
I updated schema example as it was invalid. Without checking examples manually, we should not put them in docs as they are not helpful and only confuse

**Pull request merged**